pkg/e2e/actuators/actuators.go

```diff
@@ -0,0 +1,411 @@
+package actuators
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	e2e "github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/framework"
+	mapiv1beta1 "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	kpolicyapi "k8s.io/api/policy/v1beta1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	workerRoleLabel = "node-role.kubernetes.io/worker"
+	waitShort       = 1 * time.Minute
+	waitMedium      = 3 * time.Minute
+	waitLong        = 10 * time.Minute
+	retryMedium     = 5 * time.Second
+)
+
+var nodeDrainLabels = map[string]string{
+	workerRoleLabel:      "",
+	"node-draining-test": string(uuid.NewUUID()),
+}
+
+func machineFromMachineset(machineset *mapiv1beta1.MachineSet) *mapiv1beta1.Machine {
+	randomUUID := string(uuid.NewUUID())
+
+	machine := &mapiv1beta1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: machineset.Namespace,
+			Name:      "machine-" + randomUUID[:6],
+			Labels:    machineset.Labels,
+		},
+		Spec: machineset.Spec.Template.Spec,
+	}
+	if machine.Spec.ObjectMeta.Labels == nil {
+		machine.Spec.ObjectMeta.Labels = map[string]string{}
+	}
+	for key := range nodeDrainLabels {
+		if _, exists := machine.Spec.ObjectMeta.Labels[key]; exists {
+			continue
+		}
+		machine.Spec.ObjectMeta.Labels[key] = nodeDrainLabels[key]
+	}
+	return machine
+}
+
+func isNodeReady(node *corev1.Node) bool {
+	for _, c := range node.Status.Conditions {
+		if c.Type == corev1.NodeReady {
+			return c.Status == corev1.ConditionTrue
+		}
+	}
+	return false
+}
+
+func waitUntilNodesAreReady(client runtimeclient.Client, listOpt *runtimeclient.ListOptions, nodeCount int) error {
```

Comment:
I think we should pass in a `Context` here - start with that in mind so that we can do cancellation.

Reply:
We should have to set `BeforeEach` for each `g.Describe` like this (most of the code can be put into a function and called, yet still needs to be repeated):

```go
	var describeContext context.Context

	g.BeforeEach(func() {
		var cancel context.CancelFunc

		g.By("Creating context")

		stopCh := signals.SetupSignalHandler()
		describeContext, cancel = context.WithCancel(context.Background())
		go func() {
			select {
			case <-stopCh:
				cancel()
			case <-describeContext.Done():
			}
		}()
	})
```

Reply:
Let's discuss this and expectations in the next iteration of the tests. Right now, I am not sure how we should approach this.